src/stream/src/executor/stream_reader.rs

```diff
@@ -25,32 +25,35 @@ use risingwave_connector::source::{BoxSourceWithStateStream, StreamChunkWithStat
 use tokio::sync::mpsc::UnboundedReceiver;
 
 use crate::executor::error::{StreamExecutorError, StreamExecutorResult};
-use crate::executor::Barrier;
-
-type SourceReaderMessage = StreamExecutorResult<Either<Barrier, StreamChunkWithState>>;
-type SourceReaderArm = BoxStream<'static, SourceReaderMessage>;
-type SourceReaderStreamInner =
-    SelectWithStrategy<SourceReaderArm, SourceReaderArm, impl FnMut(&mut ()) -> PollNext, ()>;
-
-pub(super) struct SourceReaderStream {
-    inner: SourceReaderStreamInner,
+use crate::executor::{Barrier, Message};
+
+type ExecutorMessageStream = BoxStream<'static, StreamExecutorResult<Message>>;
+type ReaderStreamData = StreamExecutorResult<Either<Message, StreamChunkWithState>>;
+type ReaderArm = BoxStream<'static, ReaderStreamData>;
+type ReaderStreamWithPauseInner =
+    SelectWithStrategy<ReaderArm, ReaderArm, impl FnMut(&mut PollNext) -> PollNext, PollNext>;
+
+/// `ReaderStreamWithPause` merges two streams, with one of them receiving barriers. When the
+/// barrier requires the data stream
```

Comment:
```suggestion
/// barrier requires the data stream, ...
```

Reply:
What about mentioning which side will be biased when `BIASED` is `true` somewhere?